### PR TITLE
Fix doubled warnings being printed for manifest warnings when loading ports.

### PR DESCRIPTION
**Pull request by BillyONeal**
Resolves https://github.com/microsoft/vcpkg/issues/24084
Resolves https://github.com/microsoft/vcpkg/issues/34095

The root cause of this bug is the separation between the BaselineProvider and the PortfileProvider. For the built in registry, to determine what the baseline version for the builtin registry is, we must load and parse the requested port to read what version is declared in its JSON/CONTROL file. However, the interface by which the registry tells the calling code what to do passed only the directory with the port inside:

```c++
    struct RegistryEntry
    {
        // This forces BuiltinRegistry to parse the JSON/CONTROL
        virtual ExpectedL<View<Version>> get_port_versions() const = 0;
 
        // This forces the caller to parse the JSON/CONTROL
        virtual ExpectedL<PortLocation> get_version(const Version& version) const = 0;
 
        virtual ~RegistryEntry() = default;
    };
```

Thus, callers were forced to parse the files *again*, causing reemission of any warnings like bad SPDX expressions.

The fix is to push the parsing of the port files into `RegistryEntry` so that the builtin registry can return a clone of the file it already parsed, rather than getting it to be parsed again.

### CODE REVIEW

File: src/vcpkg/commands.format-manifest.cpp [hunk 1]
```diff
@@ -41,7 +41,8 @@ namespace
         auto res = serialize_manifest(*data.scf);
 
         // reparse res to ensure no semantic changes were made
-        auto maybe_reparsed = SourceControlFile::parse_project_manifest_object(StringView{}, res, null_sink);
+        auto maybe_reparsed =
```
Comment (BillyONeal) [inline src/vcpkg/commands.format-manifest.cpp:44]: This is a drive-by fix because `format-manifest` crashes in debug builds because origin is empty.

File: src/vcpkg/portfileprovider.cpp [hunk 2]
```diff
@@ -174,40 +149,30 @@ namespace vcpkg
...
                         }
                         else
                         {
-                            return std::move(maybe_scfl)
-                                .error()
-                                .append_raw('\n')
-                                .append_raw(NotePrefix)
-                                .append(msgWhileLoadingPortVersion, msg::version_spec = version_spec)
-                                .append_raw('\n');
+                            get_global_metrics_collector().track_define(DefineMetric::VersioningErrorVersion);
```
Comment (BillyONeal) [inline src/vcpkg/portfileprovider.cpp:162]: This DefineMetric will now fire in more places now; 'the version DB says the version is one way but upon inspection it's something else' seems more appropriate for it? Would be happy to just delete it too /cc @vicroms

File: src/vcpkg/sourceparagraph.cpp [hunk 3]
```diff
@@ -1339,6 +1339,8 @@ namespace vcpkg
         {
             ret.feature_paragraphs.push_back(std::make_unique<FeatureParagraph>(*feat_ptr));
         }
+
+        ret.extra_features_info = extra_features_info;
```
Comment (BillyONeal) [inline src/vcpkg/sourceparagraph.cpp:1343]: I really wish this thing was just copyable, but it has this clone function and I'm being consistent with it.

File: src/vcpkg/registries.cpp [hunk 4]
```diff
@@ -444,12 +446,23 @@ namespace
...
+        explicit FilesystemRegistryEntry(const ReadOnlyFilesystem& fs,
+                                         StringView port_name,
+                                         std::vector<FilesystemVersionDbEntry>&& version_entries)
+            : fs(fs), port_name(port_name.data(), port_name.size())
+        {
+            for (auto&& version_entry : version_entries)
+            {
+                port_versions.push_back(std::move(version_entry.version.version));
+                version_paths.push_back(std::move(version_entry.p));
+            }
```
Comment (BillyONeal) [inline src/vcpkg/registries.cpp:458]: This was moved from line 855 to make `FilesystemRegistryEntry` actually be usable when the constructor completes.

**Review by ras0219-msft — APPROVED** [review]

File: src/vcpkg/registries.cpp [hunk 5]
```diff
@@ -401,27 +401,29 @@ namespace
...
         {
         }
 
         ExpectedL<View<Version>> get_port_versions() const override { return View<Version>{&version, 1}; }
-        ExpectedL<PortLocation> get_version(const Version& v) const override
+        ExpectedL<SourceControlFileAndLocation> try_load_port(const Version& v) const override
         {
             if (v == version)
             {
-                return PortLocation{root, "git+https://github.com/Microsoft/vcpkg#ports/" + name};
```
Comment (BillyONeal) [inline src/vcpkg/registries.cpp:414]: I need to double check this location is still set
